code/modules/food_and_drinks/kitchen_machinery/processor.dm

```diff
@@ -144,16 +169,14 @@
 			log_admin("DEBUG: [movable_input] in processor doesn't have a suitable recipe. How did it get in there? Please report it immediately!!!")
 			continue
 		total_time += recipe.time
-	var/offset = prob(50) ? -2 : 2
-	animate(src, pixel_x = pixel_x + offset, time = 0.2, loop = (total_time / rating_speed)*5) //start shaking
+	shake_for(total_time / rating_speed)
 	sleep(total_time / rating_speed)
 	for(var/atom/movable/O in processor_contents)
 		var/datum/food_processor_process/P = PROCESSOR_SELECT_RECIPE(O)
 		if (!P)
 			log_admin("DEBUG: [O] in processor doesn't have a suitable recipe. How do you put it in?")
 			continue
 		process_food(P, O)
-	pixel_x = base_pixel_x //return to its spot after shaking
 	processing = FALSE
 	visible_message(span_notice("\The [src] finishes processing."))
```

Comment:
Instead of a sleep, this should probably be put in a separate proc and be called using an `addtimer()` with a CALLBACK in it.

Reply:
Out of curiosity, why shouldn't sleep be used?

Reply:
Sleeps cannot be managed by the Master Controller. A sleep in a subsystem proc, in this case, process(), will sleep the entire subsystem, which is EXTRA bad for a processing subsystem.

Reply:
@Kapu1178 https://github.com/tgstation/tgstation/pull/68885/commits/69287cba2f3159aac68ac1c5ba1d9e13f9a26911